utils/clickhouse.cc

```diff
@@ -0,0 +1,82 @@
+/**
+ *		Tempesta FW
+ *
+ * Copyright (C) 2024 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+#include <iostream>
+
+#include <clickhouse/base/socket.h>
+#include <clickhouse/client.h>
+
+#include "clickhouse.hh"
+
+
+constexpr size_t MAX_MSEC = 100;
+constexpr size_t MAX_EVENTS = 1000;
+
+static auto
+now_ms()
+{
+	return std::chrono::duration_cast<std::chrono::milliseconds>(
+			std::chrono::system_clock::now().time_since_epoch())
+		.count();
+}
+
+TfwClickhouse::TfwClickhouse(std::string host, std::string table_name,
+			     std::string user, std::string password,
+			     std::shared_ptr<clickhouse::Block> (*cb)())
+{
+	auto opts = clickhouse::ClientOptions();
+
+	block_callback_ = cb;
+	table_name_ = table_name;
+
+	opts.SetHost(std::move(host));
```

Comment:
This is from configuration options, so only first thread will succeed. We can use `std::move` only if the source isn't used anymore.

Reply:
@krizhanovsky here a copy of a `host` string is used and `host` is not used any more, so stealing a value by `std::move` is okay. 
I see another mistake - SetHost doesnot accept an rvalue, so move semantics doesnot make sense

Reply:
Fixed.